private/src/gst/gstavbvideosrc.cpp

```diff
@@ -129,21 +172,50 @@ gst_avbvideosrc_change_state(GstElement *element, GstStateChange transition)
 {
     GstAvbVideoSrc *avbvideosrc = GST_AVBVIDEOSRC(element);
     GstStateChangeReturn ret = GST_STATE_CHANGE_SUCCESS;
+    GstCaps *caps_event = NULL;
 
     if (transition == GST_STATE_CHANGE_NULL_TO_READY) {
         ias_avbvideobridge_result r;
 
-        avbvideosrc->receiver = ias_avbvideobridge_create_receiver("MpegTs_Receiver",
+        avbvideosrc->receiver = ias_avbvideobridge_create_receiver("receiver",
                 avbvideosrc->stream_name);
+
+        if (avbvideosrc->stream_type == TYPE_MPEG_TS) {
+            caps_event = gst_caps_new_simple("video/mpegts",
+                                             "packetsize", G_TYPE_INT, MPEG_TS_PACKET_SIZE,
+                                             "systemstream", G_TYPE_BOOLEAN, TRUE,
+                                             NULL);
+        } else if (avbvideosrc->stream_type == TYPE_RTP_H264) {
+            caps_event = gst_caps_new_simple("application/x-rtp",
+                                             "media", G_TYPE_STRING, "video",
+                                             "clock-rate", G_TYPE_INT, 90000,
+                                             "encoding-name", G_TYPE_STRING, "H264",
+                                             NULL);
+        } else {
+            GST_ERROR_OBJECT(avbvideosrc, "Invalid stream type");
+            return GST_STATE_CHANGE_FAILURE;
+        }
+
+        if (!caps_event && !(gst_base_src_set_caps(GST_BASE_SRC(avbvideosrc), caps_event))) {
+            GST_ERROR_OBJECT(avbvideosrc, "Failed to set caps on the source pad");
+            return GST_STATE_CHANGE_FAILURE;
+        }
+
         if (!avbvideosrc->receiver) {
```

Comment:
Here, this code should be near the `ias_avbvideobridge_create_receiver` call.

Reply:
Done